src/index/IndexImpl.Text.cpp

```diff
@@ -140,17 +153,21 @@ void IndexImpl::addTextFromContextFile(const string& contextFile,
   readConfiguration();
   vocab_.readFromFile(onDiskBase_ + VOCAB_SUFFIX);
 
+  scoreData_ = {vocab_.getLocaleManager(), textScoringMetric_,
+                bAndKParamForTextScoring_};
+
   // Build the text vocabulary (first scan over the text records).
-  LOG(INFO) << "Building text vocabulary ..." << std::endl;
   size_t nofLines =
-      processWordsForVocabulary(contextFile, addWordsFromLiterals);
-
+      processWordsForVocabulary(wordsAndDocsFile.first, addWordsFromLiterals);
+  // Calculate the score data for the words
+  scoreData_.calculateScoreData(wordsAndDocsFile.second, addWordsFromLiterals,
+                                textVocab_, vocab_);
```

Comment:
Can we still use the explicitly precalculated scores that are stored in the text file with your new approach? In that case you could skip this case here.